File: lib/organize_file.rb
def organize_file(log_files)
  dic = {}
  temp = []
  new_filename = ''
  sorted = log_files.sort_by { |x| x[2] }
  sorted.map do |log_file|
    log_file.each_with_index do |file, index|
      if index == 1
        if dic[file]
          seq = dic[file].length
          count = (seq + 1).to_s
          suffix = format('%02d', count)
          filename = file + suffix
        else
          count = index.to_s
          suffix = format('%02d', count)
          filename = file + suffix
          dic[file] = []
        end
        dic[file] << filename
        temp[0] = filename
      end
      if index.zero?
        ext = file[-4, file.size]
        temp[1] = ext
      end
      new_filename = temp.join('')
    end
    new_filename
  end
end

File: lib/organizeFile.ts
function organizeFile(logFiles: string[][]) {
  let memo: { [key: string]: string[] } = {}
  let temp: string[] = []
  let newFilename = ''
  let result = logFiles.sort(function (a, b) {
    if (a[2] === b[2]) {
      return 0
    }
    return (a[2] < b[2]) ? -1 : 1
  })
  .map(function (logFile) {
    logFile.forEach(function(file, index) {
      let filename;
      switch (index) {
        case 0:
          let ext = file.slice(-4, file.length)
          temp[1] = ext
        break;
        case 1:
          if (memo[file]) {
            let seq = memo[file].length
            let count = (seq + 1).toString()
            let suffix = Array(Math.max(3 - count.length, 0)).join('0') + count
            filename = file + suffix
          } else {
            let count = index.toString()
            let suffix = Array(Math.max(3 - count.length, 0)).join('0') + count
            filename = file + suffix
            memo[file] = []
          }
          memo[file].push(filename)
          temp[0] = filename
        break;

        default:
          break;
      }
      newFilename = temp.join('')
    });
    return newFilename
  })
  return result
}

export default organizeFile
